service/history/workflow/update/update.go

```diff
@@ -322,6 +330,11 @@ func (u *Update) Admit(
 	if u.state != stateCreated {
 		return nil
 	}
+	if err := u.checkLimits(req); err != nil {
+		// Remove the update from the registry immediately.
+		u.onComplete()
+		return err
```

Comment:
Right now, this isn't executed yet.

Reply:
It is good to have it there right away though.